File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:

        maxLength = 0
        dp = [ 1 for _ in range(len(nums)) ]
        
        for i in range(len(nums)):
            for j in range(i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], dp[j]+1)
            
        return max(dp)

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {

        int[] dp = new int[nums.length];
        Arrays.fill(dp, 1);

        int maxLength = 0;

        for(int i=0; i<nums.length; i++){
            for(int j=0; j<i; j++){
                if(nums[i] >  nums[j]){
                    dp[i] = Math.max(dp[i], dp[j]+1);
                }
            }
            maxLength = Math.max(maxLength, dp[i]);
        }

        return maxLength;
    }
}